src/java.base/share/classes/jdk/internal/classfile/impl/ClassReaderImpl.java

```diff
@@ -189,45 +189,73 @@ public BootstrapMethodEntryImpl bootstrapMethodEntry(int index) {
         return bsmEntries().get(index);
     }
 
+    private static IllegalArgumentException outOfBoundsError(IndexOutOfBoundsException cause) {
+        return new IllegalArgumentException("Reading beyond classfile bounds", cause);
+    }
+
     @Override
     public int readU1(int p) {
-        return buffer[p] & 0xFF;
+        try {
+            return buffer[p] & 0xFF;
+        } catch (IndexOutOfBoundsException e) {
```

Comment:
This and all other catch blocks introduced in this change can be changed to the specific exception type `ArrayIndexOutOfBoundsException`, because all these operations are dealing with only array access. If you prefer catching this more generic `IndexOutOfBoundsException`, that's fine with me.

Reply:
Yes, I prefer the generic `IndexOutOfBoundsException`, thanks.